src/blueapi/core/context.py

```diff
@@ -97,16 +114,22 @@ def plan_2(...) -> MsgGenerator:
 
         for obj in load_module_all(module):
             if is_bluesky_plan_generator(obj):
-                self.plan(obj)
+                self.register_plan(obj)
 
     def with_device_module(self, module: ModuleType) -> None:
         self.with_dodal_module(module)
 
+    def get_lazy_devices(self) -> DevicesDict:
+        return {k: v for k, v in self.devices.items() if v.lazy}
+
     def with_dodal_module(self, module: ModuleType, **kwargs) -> None:
         devices, exceptions = make_all_devices(module, **kwargs)
 
-        for device in devices.values():
-            self.device(device)
+        # for non-lazy devices, we instantiate them
+        early_devices, _ = bisect_devices_dict(devices)
```

Comment:
This seems to be the only place you use the bisected dictionaries, in which case, is there any point in bisecting and storing two dictionaries? Could you just have one big dictionary and filter for eager devices here?